src/ASM/ASMs2Dmx.C

```diff
@@ -277,60 +278,59 @@ bool ASMs2Dmx::generateFEMTopology ()
 
   size_t iel = 0, inod = 0;
   for (auto& it : m_basis)
-    for (i2 = 0; i2 < it->numCoefs_v(); i2++)
-      for (i1 = 0; i1 < it->numCoefs_u(); i1++)
+    for (int i2 = 0; i2 < it->numCoefs_v(); i2++)
+      for (int i1 = 0; i1 < it->numCoefs_u(); i1++)
       {
         myNodeInd[inod].I = i1;
         myNodeInd[inod].J = i2;
         myMLGN[inod++]    = ++gNod;
       }
 
-  int firstItgNode = this->getFirstItgElmNode();
-  int nElmNode = std::accumulate(elem_size.begin(), elem_size.end(), 0);
+  const size_t firstItgElmNode = this->getFirstItgElmNode();
+  const size_t nElmNode = std::accumulate(elem_size.begin(), elem_size.end(), 0);
 
   // Create nodal connectivities for bases
+  auto&& enumerate_elem = [this,&iel](const Go::SplineSurface& spline,
```

Comment:
since `iel` is just an int and is not updated by the lambda, do we need to catch the reference here?

Reply:
heh. this is a common question and a common mistake to make. if we do not capture by reference, we capture the value when the lambda is instanced, and it will have that value in all subsequent calls. so yes, i do need to capture by reference.

Reply:
really? I though it was captured when invoked. That's a dangerous difference I was unaware of. Hopefully I have not introduced such lambdas elsewhere..

Thank's for the lesson ;)